swap/swap.go

```diff
@@ -537,8 +536,38 @@ func (s *Swap) getContractOwner(ctx context.Context, address common.Address) (co
 	return contr.Issuer(nil)
 }
 
-// Deploy deploys the Swap contract
-func (s *Swap) Deploy(ctx context.Context, backend swap.Backend, path string) error {
+// StartChequebook deploys a new instance of a chequebook if chequebookAddr is empty, otherwise it wil bind to an existing instance
+func (s *Swap) StartChequebook(chequebookAddr common.Address) (err error) {
+	if chequebookAddr != (common.Address{}) {
+		if err := s.BindToContractAt(chequebookAddr); err != nil {
+			return err
+		}
+		log.Info("Using the provided chequebook", "chequebookAddr", chequebookAddr)
+	} else {
+		if err := s.Deploy(context.Background(), s.backend); err != nil {
+			return err
+		}
+		log.Info("New SWAP contract deployed", "contract info", s.DeploySuccess())
+	}
+	return nil
+}
+
+// BindToContractAt binds an instance of an already existing chequebook contract at address and sets chequebookAddr
+func (s *Swap) BindToContractAt(address common.Address) (err error) {
+
+	if err := contract.ValidateCode(context.Background(), s.backend, address); err != nil {
+		return fmt.Errorf("contract validation for %v failed: %v", address, err)
+	}
+	s.contract, err = contract.InstanceAt(address, s.backend)
+	if err != nil {
+		return err
```

Comment:
since `err` is implicitly initialized, you can just `return` here

Reply:
get rid of named paramater. 
@mortelli our code policy is to write return arguments explicitly even if we have named return params

Reply:
I would like to keep the initialization of `err` in the function initialization, because otherwise, I can't do 
```golang
s.contract, err = contract.InstanceAt(address, s.backend)
```
Obviously, if it is a strict policy not to have named return parameters, I can do that, but in this case, I need 2 extra lines of code (to initialize a variable for the `contractInstance`, and assign this to `s.contract`). This was how I did it initially, but @holisticode asked me to do it in a one-liner.

Reply:
this is fine. I also like sometimes that vars are initialised,
 the policy is more about not having bare `return`-s if the function has return values. 
it makes things more readable